lapis/src/main/kotlin/org/genspectrum/lapis/model/AdvancedQueryCustomListener.kt

```diff
@@ -0,0 +1,534 @@
+                    throw BadRequestException("'$metadataValue' is not a valid boolean", e)
+                }
+            }
+
+            SequenceFilterFieldType.Date -> {
+                try {
+                    val date = LocalDate.parse(metadataValue)
+                    expressionStack.addLast(DateBetween(field.name, to = date, from = date))
+                } catch (exception: DateTimeParseException) {
+                    throw BadRequestException("'$metadataValue' is not a valid date: ${exception.message}", exception)
+                }
+            }
+
+            is SequenceFilterFieldType.DateFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.DateTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            SequenceFilterFieldType.Float -> {
+                try {
+                    expressionStack.addLast(FloatEquals(field.name, metadataValue.toDouble()))
+                } catch (e: NumberFormatException) {
+                    throw BadRequestException("'$metadataValue' is not a valid float", e)
+                }
+            }
+
+            is SequenceFilterFieldType.FloatFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.FloatTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            SequenceFilterFieldType.Int -> {
+                try {
+                    expressionStack.addLast(IntEquals(field.name, metadataValue.toInt()))
+                } catch (e: NumberFormatException) {
+                    throw BadRequestException("'$metadataValue' is not a valid integer", e)
+                }
+            }
+
+            is SequenceFilterFieldType.IntFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.IntTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
```

Comment:
Do we really want to allow unquoted regexes? We could simply decide to stick to the requirement that the value must have quotes.